app/src/main/java/com/example/sharingang/NewItemActivity.kt

```diff
@@ -17,7 +17,7 @@ class NewItemActivity : AppCompatActivity() {
     }
 
     fun createNewItem(view: View) {
-        val intent = Intent(this, MainActivity::class.java)
+        val intent = Intent()
```

Comment:
Instead of explicitly declaring that we want to launch the MainActivity, we just say that we want to return a result from this activity, ending up at whatever activity called us.

In practice, this gives us the same result, since MainActivity is the only thing launching NewItemActivity for its result, but if another activity were to do the same, explicitly using MainActivity in the intent here wouldn't return to the right place, I think.